tracer/src/Datadog.AutoInstrumentation.NativeLoader/instrumented_assembly_generator/instrumented_assembly_generator_helper.h

```diff
@@ -0,0 +1,330 @@
+        const auto instrumentedAssemblyGeneratorDir = logsDirectory / InstrumentedAssemblyGeneratorLogsFolder;
+
+        const auto processName = shared::GetCurrentProcessName();
+        const auto processId = shared::GetPID();
+        if (!processName.empty() && processId > 0)
+        {
+            const auto time = shared::GetProcessStartTime();
+            static auto processWithoutExtension = processName.substr(0, processName.find_last_of(L'.'));
+            const auto fullName = processWithoutExtension + WStr("_") + shared::ToWSTRING(processId) + WStr("_") + time;
+
+            const auto cleanedDirectoryName = GetCleanedFileName(fullName);
+            const auto instrumentedAssemblyGeneratorAppDir = instrumentedAssemblyGeneratorDir / cleanedDirectoryName;
+            CreateNecessaryFolders(instrumentedAssemblyGeneratorAppDir);
+            instrumentedAssemblyGeneratorOutputFolder = instrumentedAssemblyGeneratorAppDir;
+        }
+    }
+    catch (const std::exception& e)
+    {
+        Log::Error("GetInstrumentedAssemblyGeneratorCurrentProcessFolder: failed to get instrumented log path. ", e.what());
+    }
+    catch (...)
+    {
+        Log::Error("GetInstrumentedAssemblyGeneratorCurrentProcessFolder: failed to get instrumented log path.");
+    }
+    return instrumentedAssemblyGeneratorOutputFolder;
+}
+
+// Check if we should use the instrumented assembly generator to generate files
+inline bool IsInstrumentedAssemblyGeneratorEnabled()
+{
+    try
+    {
+        const auto isInstrumentedAssemblyGeneratorEnabled = shared::GetEnvironmentValue(cfg_instrumentation_verification_env);
+        if (isInstrumentedAssemblyGeneratorEnabled == WStr("1") ||
+            isInstrumentedAssemblyGeneratorEnabled == WStr("true"))
+        {
+#if _WIN32
+            if (const auto path = GetInstrumentedAssemblyGeneratorCurrentProcessFolder(); !path.empty())
+            {
+                Log::Info("Instrumentation Verification log is enabled. Output folder is: ", path);
+                return true;
+            }
+#else 
+            Log::Warn("Instrumentation Verification is currently only supported on Windows and will be disabled.");
+#endif
+
+        }
+    }
+    catch (...)
+    {
+    }
+    return false;
+}
+
+inline std::tuple<HRESULT, shared::WSTRING, shared::WSTRING> GetModuleNameAndMvid(const ComPtr<IMetaDataImport>& metadataImport)
+{
+    ULONG pchName;
+    GUID pmvid;
+    WCHAR szName[path_length_limit];
+    auto hr = metadataImport->GetScopeProps(szName, path_length_limit, &pchName, &pmvid);
+    if (FAILED(hr))
+    {
+        return {hr, nullptr, nullptr};
+    }
+    shared::WSTRING name = szName;
+    shared::WSTRING mvid = shared::ToWSTRING(AnsiStringFromGuid(pmvid));
+    return {hr, name, mvid};
+}
+
+inline bool ShouldSkipCopyModule(const shared::WSTRING& moduleName)
+{
+    if (shared::WStringStartWithCaseInsensitive(moduleName, WStr("RefEmit_"))) return true;
+    return false;
```

Comment:
nit: you can just 
`return shared::WStringStartWithCaseInsensitive(moduleName, WStr("RefEmit_"));`

Reply:
Fixed in ebde2f57c8ef03fd167862f981abd5c4958c222a